src/commands/getMarkRange.ts

```diff
@@ -0,0 +1,46 @@
+import { Mark, MarkType, ResolvedPos } from "prosemirror-model"
+import { MarkRange } from "../core/types.js"
+
+// TODO this doesn't handle selections _across_ nodes
```

Comment:
@makenosound are you able to help me understand what doesn't work here? I've tried adding/removing marks across nodes and it seems to work fine.

Reply:
I can’t quite remember off the top of my head, but from my vague read of the code and my vague memories I _think_ the deal is that:

1. This code tries to find the full range for a `mark` — i.e., where it starts and where it ends
2. If a mark crosses multiple nodes it doesn’t look beyond the starting node

I don’t think this is really a problem in practice, but I _believe_ it’s possible for a mark to span across multiple nodes in _some_ situations. Perhaps worth noting it in a comment as a limitation of the implementation rather than a thing that has to be fixed?